### PR TITLE
IGNITE-15042: fixed SqlSystemViewsSelfTest#testCachesViews

### PR DESCRIPTION
**Pull request by Sega76**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java [hunk 1]
```diff
@@ -1567,7 +1567,6 @@ public void testCachesViews() throws Exception {
             "ON C.CACHE_NAME = CG.CACHE_GROUP_NAME WHERE C.CACHE_NAME like 'cache%' " +
             "AND C.CACHE_MODE = CG.CACHE_MODE " +
             "AND C.ATOMICITY_MODE = CG.ATOMICITY_MODE " +
-            "AND COALESCE(C.AFFINITY, '-') = COALESCE(CG.AFFINITY, '-') " +
```
Comment (timoninmaxim) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java:1570]: What about to fix Affinity.toString instead?

Comment (Sega76) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java:1570]: thought about it in the first commit
(https://github.com/apache/ignite/pull/9215/commits/94ab8aa0f345acb5901e9e2bf8abf98849894101), 
but this change looks unexpected and better in a separate ticket

Comment (timoninmaxim) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java:1570]: I think that fix of affinity.toString will fix this issue too. And toString is a core problem, so let's fix it instead, if it possible.

Comment (nizhikov) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java:1570]: AFAIK - `lastWakeUpTime` is changes between to consequent `toString` invocation.
I don't think we should remove it from `Affinity#toString`.

Comment (timoninmaxim) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java:1570]: Sorry, I don't found lastWakeUpTime in affinity functions, could you please provide a link? I see that AffinityFunction has static fields only, like  `partitions`, `backups`.

The problem I saw that RendevouzAffinityFunction has Ignite instance in toString. And we return it to user when it ask for affintiy function from view. Looks incorrect, isn't it?

**Review by alex-plekhanov — APPROVED** [review]

**Pull request closed without merging**